homeassistant/components/reolink/binary_sensor.py

```diff
@@ -0,0 +1,177 @@
+    supported: Callable[[Host, int | None], bool] = lambda host, ch: True
+
+
+BINARY_SENSORS = (
+    ReolinkBinarySensorEntityDescription(
+        key="motion",
+        name="Motion",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda api, ch: api.motion_detected(ch),
+    ),
+    ReolinkBinarySensorEntityDescription(
+        key=FACE_DETECTION_TYPE,
+        name="Face",
+        icon="mdi:face-recognition",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda api, ch: api.ai_detected(ch, FACE_DETECTION_TYPE),
+        supported=lambda api, ch: api.ai_supported(ch, FACE_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorEntityDescription(
+        key=PERSON_DETECTION_TYPE,
+        name="Person",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda api, ch: api.ai_detected(ch, PERSON_DETECTION_TYPE),
+        supported=lambda api, ch: api.ai_supported(ch, PERSON_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorEntityDescription(
+        key=VEHICLE_DETECTION_TYPE,
+        name="Vehicle",
+        icon="mdi:car",
+        icon_off="mdi:car-off",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda api, ch: api.ai_detected(ch, VEHICLE_DETECTION_TYPE),
+        supported=lambda api, ch: api.ai_supported(ch, VEHICLE_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorEntityDescription(
+        key=PET_DETECTION_TYPE,
+        name="Pet",
+        icon="mdi:dog-side",
+        icon_off="mdi:dog-side-off",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda api, ch: api.ai_detected(ch, PET_DETECTION_TYPE),
+        supported=lambda api, ch: api.ai_supported(ch, PET_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorEntityDescription(
+        key="visitor",
+        name="Visitor",
+        icon="mdi:bell-ring-outline",
+        icon_off="mdi:doorbell",
+        value=lambda api, ch: api.visitor_detected(ch),
+        supported=lambda api, ch: api.is_doorbell_enabled(ch),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Reolink IP Camera."""
+    reolink_data: ReolinkData = hass.data[DOMAIN][config_entry.entry_id]
+    host = reolink_data.host
+
+    entities = []
+    for channel in host.api.channels:
+        for entity_description in BINARY_SENSORS:
+            if not entity_description.supported(host, channel):
+                continue
+
+            entities.append(
+                ReolinkBinarySensorEntity(
+                    reolink_data,
+                    channel,
+                    entity_description,
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class ReolinkBinarySensorEntity(ReolinkCoordinatorEntity, BinarySensorEntity):
+    """An implementation of a base binary-sensor class for Reolink IP camera motion sensors."""
```

Comment:
This docs block is to long (goes over the 88 char limit).

Reply:
done